src/controllers/OnSearchController.js

```diff
@@ -1,16 +1,21 @@
 import LoggingService from '../services/LoggingService';
 import SearchService from '../services/SearchService';
+import genericResponse from './GenericResponse';
+import authVerifier from '../utilities/SignVerify/AuthHeaderVerifier';
 
-const onSearch = (req, res) => {
+const BecknGateway = 'BG';
+const onSearch = async (req, res) => {
   const logger = LoggingService.getLogger('OnSearchController');
   logger.debug(`on_search called with ${JSON.stringify(req.body)}`);
-  SearchService.storeSearchResult(req.body);
-  res.send({
-    message: {
-      ack: {
-        status: 'ACK',
-      },
-    },
+
+  const publicKey = await genericResponse.getPublicKey(BecknGateway);
```

Comment:
Can we move getPublicKey to something like LookupService.?

Reply:
Done.